src/supported-operations/thin-edge-supported-operations.md

```diff
@@ -0,0 +1,118 @@
+# Operation Configuration Examples
+
+All operation configuration files use the `toml` format.
+
+```toml
+[exec]
+  # Exec configuration if the operation requires command execution
+  # Required
+  command = "echo"
+  # Optional
+  root = true
+
+[mqtt]
+  # MQTT configuration if the operation requires MQTT communication, e.g. forwarding message JSON on the bus
+  topic = "tedge/logs"
+
+[extras]
+  # Additional configuration if the operation requires additional configuration
+  log_type = ["error"]
+```
+
+* If the operation doesn't require additional configuration, an empty `toml` file can be used.
+* Basic tables names are fixed and are `exec`, `mqtt` and `extras`. Additional tables can be added as needed.
+* Tables `[exec]` or `[mqtt]` must be present if config is not empty.
+* Tables `[exec]` and `[mqtt]` are mutually exclusive and only one of them can be used.
+
+## Example 1
+
+Given operation `restart`, the operation file would be created in `/etc/tegde/operations/c8y` with the filename `c8y_Restart` so the full path would be `/etc/tegde/operations/c8y/c8y_Restart`.
+Assuming `restart` operation requires no additional parameters as well as can be executed directly in the executor (assume a mapper) no additional configuration is required.
+
+An empty `toml` file is still valid `toml` file and can be used to indicate no additional configuration is required.
+
+Given a mapper the flow would be as follows:
+
+1. Mapper reads directory for the cloud which it supports i.e. `/etc/tegde/operations/c8y`.
+2. Mapper finds the operation file for the operation `restart` in the directory `c8y` with filename `c8y_Restart` which is the only operation.
+3. Mapper takes the filename as operation name and reads the operation file.
+4. Empty operation file means there are no additional configuration parameters and mapper is ready to send supported operations message to c8y which contains the list of supported operations i.e. `c8y_Restart`.
+5. Cloud operator wants restart the device therefore they send the operation message to the device which mapper interprets as a restart operation and mapper executes restart.
+
+## Example 2 (special case)
+
+Given operation `c8y_LogfileRequest`, the operation file would be created in `/etc/tegde/operations/c8y` with the filename `c8y_LogfileRequest` so the full path would be `/etc/tegde/operations/c8y/c8y_LogfileRequest`.
+Operation `c8y_LogfileRequest` requires additional parameter `log_type` to be set.
+
+With the operation file the following structure is written:
+
+```toml
+[exec]
+
+command = "/etc/tedge/plugins/c8y_LogfileRequest"
+user = "root"
+
+[init]
+topic = "c8y/s/us"
+message = "116,error"
+
+[extras]
+  # Additional configuration if the operation requires additional configuration
+  log_type = ["error"]
```

Comment:
I'm wondering how the mapper would use just this `log_type` information without knowing what SmartREST message code to send it with(`118` in this case). So, the proposal with the `init` block looks better.

Reply:
This is why I think it would be an endless process if we start to model the operations and the cloud-specific protocols.

* Sending such messages inside the operation plugin is easy and not contrived, the plugin being already cloud & operation specific.
* If we model each and every operation in these configuration files, then why not directly providing the feature ourselves?